src/v/cluster/config_frontend.cc

```diff
@@ -87,10 +88,36 @@ ss::future<config_frontend::patch_result> config_frontend::do_patch(
           .errc = errc::waiting_for_recovery, .version = config_version_unset};
     }
 
-    auto data = cluster_config_delta_cmd_data();
+    // do not transform values if the cluster is in the middle of an upgrade.
+    // our aliases might be primary names for the other nodes
+    auto in_mixed_state = [&] {
+        auto& ft = _features.local();
+        return ft.get_active_version() != ft.get_latest_logical_version();
+    }();
```

Comment:
nit: seems like this could just be a bool, rather than a lambda()

Reply:
yea, it was to wrap much worse looking code before i transplanted it here. Will change